provider/src/batchProcessor.ts

```diff
@@ -38,6 +43,18 @@ export const combineResponses = (
     // Otherwise, look up the response from the provider
     const providerResponse = responseMap.get(processed.request.id)
 
+    // If provider response is found and the request is marked as cacheable, store it
+    if (providerResponse && processed.shouldCache) {
+      storeResponseInCache(
```

Comment:
I see you are not "awaiting" here. It may be a good idea to not slow down the request but I want to make sure this is what you are trying to acheve!